assets/src/edit-story/components/panels/elementAlignment.js

```diff
@@ -0,0 +1,333 @@
+  const isDistributionEnabled = isFill || selectedElements.length < 3;
+
+  const handleAlignLeft = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetX = 0;
+      if (rotationAngle) {
+        const { width: newWidth } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetX = (newWidth - width) / 2;
+      }
+      return {
+        x: boundRect.startX + offSetX,
+      };
+    });
+  };
+
+  const handleAlignCenter = () => {
+    const centerX = (boundRect.endX + boundRect.startX) / 2;
+    onSetProperties((properties) => {
+      const { width } = properties;
+      return {
+        x: centerX - width / 2,
+      };
+    });
+  };
+
+  const handleAlignRight = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetX = 0;
+      if (rotationAngle) {
+        const { width: newWidth } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetX = (newWidth - width) / 2;
+      }
+      return {
+        x: boundRect.endX - width - offSetX,
+      };
+    });
+  };
+
+  const handleAlignTop = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetY = 0;
+      if (rotationAngle) {
+        const { height: newHeight } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetY = (newHeight - height) / 2;
+      }
+      return {
+        y: boundRect.startY + offSetY,
+      };
+    });
+  };
+
+  const handleAlignMiddle = () => {
+    const centerY = (boundRect.endY + boundRect.startY) / 2;
+    onSetProperties((properties) => {
+      const { height } = properties;
+      return {
+        y: centerY - height / 2,
+      };
+    });
+  };
+
+  const handleAlignBottom = () => {
+    onSetProperties((properties) => {
+      const { x, y, width, height, rotationAngle } = properties;
+      let offSetY = 0;
+      if (rotationAngle) {
+        const { height: newHeight } = calcRotatedObjectPositionAndSize(
+          rotationAngle,
+          x,
+          y,
+          width,
+          height
+        );
+        offSetY = (newHeight - height) / 2;
+      }
+      return {
+        y: boundRect.endY - height - offSetY,
+      };
+    });
+  };
+
+  const handleHorizontalDistribution = () => {
+    const offSetWidth = dataPixels(
+      (boundRect.endX - boundRect.startX) / (selectedElements.length - 1)
+    );
+    onSetProperties((properties) => {
+      const { id, x, y, width, height, rotationAngle } = properties;
+      const elementIndex = selectedElements.findIndex((item) => item.id === id);
```

Comment:
Hmm. Is the distribution order the same as the selection order? Seems a bit unlikely. I think we should sort `selectedElements` here by `x`.

Reply:
Is this still in progress?

Reply:
I'm actually a bit doubtful that the current way of "center alignment" is the right one here. I'm wondering if we instead should distribute the empty space. E.g. `commonSpace = sum(frameWidth)/count; delta = space[i] - commonSpace;` - that sort  of thing.

Reply:
Was working on issue 232, will push commit for this solution this evening.

Reply:
hmm... Yes. Actually that makes sense for me. Sot the space between each object should be equal